MovingBlocks/Terasology
# refactor(netork)!: rewrite ping logic

### Contains

Started off with some boy-scout rule improvements while looking at the code, and ended in a bigger experiment on consolidating (unneccessary?) complexity in the logic around pings and ping subscriptions.

Also added a bunch of docstrings to improve the comprehensibility of the code.

### How to test

Start a multiplayer game with multiple connected clients and ensure that the ping information in the UI still looks reasonable.

### Outstanding before merging

n/a
